packages/contracts/contracts/plugin/examples/count/CounterV2PluginManager.sol

```diff
@@ -0,0 +1,163 @@
+        if (oldVersion[0] == 1) {
+            (_newVariable) = abi.decode(data, (uint256));
+        }
+
+        // TODO: Shall we leave it here or make devs call `upgrade` from our abstract factory
+        // Just a way of reinforcing...
+        // TODO1: proxy needs casting to UUPSSUpgradable
+        // TODO2: 2nd line needs casting to CounterV2
+        // proxy.upgradeTo(getImplementationAddress());
+        CounterV2(proxy).setNewVariable(_newVariable);
+    }
+
+    // TODO: WOULD THIS NEED dao as well to be passed
+    function getInstallPermissions(bytes memory data)
+        external
+        view
+        virtual
+        override
+        returns (RequestedPermission[] memory permissions, string[] memory helperNames)
+    {
+        address _multiplyHelper = abi.decode(data, (address));
+
+        permissions = new RequestedPermission[](_multiplyHelper == address(0) ? 2 : 3);
+        helperNames = new string[](1);
+
+        // Allows plugin Count to call execute on DAO
+        permissions[0] = createPermission(
+            BulkPermissionsLib.Operation.Grant,
+            DAO_PLACEHOLDER,
+            PLUGIN_PLACEHOLDER,
+            NO_ORACLE,
+            keccak256("EXEC_PERMISSION")
+        );
+
+        // Allows DAO to call Multiply on plugin Count
+        permissions[1] = createPermission(
+            BulkPermissionsLib.Operation.Grant,
+            PLUGIN_PLACEHOLDER,
+            DAO_PLACEHOLDER,
+            NO_ORACLE,
+            counterBase.MULTIPLY_PERMISSION_ID()
+        );
+
+        // MultiplyHelper could be something that dev already has it from outside
+        // which mightn't be a aragon plugin. It's dev's responsibility to do checks
+        // and risk whether or not to still set the permission.
+        if (_multiplyHelper == address(0)) {
+            // Allows Count plugin to call MultiplyHelper's multiply function.
+            permissions[2] = createPermission(
+                BulkPermissionsLib.Operation.Grant,
+                0, // Index from relatedContracts (multiplyHelper)
+                PLUGIN_PLACEHOLDER,
+                NO_ORACLE,
+                multiplyHelperBase.MULTIPLY_PERMISSION_ID()
+            );
+        }
+
+        helperNames[0] = "MultiplyHelper";
+    }
+
+    // TODO: will this need `proxy` address as well ?
+    function getUpdatePermissions(uint16[3] calldata oldVersion, bytes memory data)
+        external
+        view
+        virtual
+        override
+        returns (
+            RequestedPermission[] memory permissions,
+            string[] memory /* helperNames */
+        )
+    {
+        address whoCanCallMultiply = abi.decode(data, (address));
+
+        permissions = new RequestedPermission[](2);
+
+        // Now, revoke permission so dao can't call anymore this multiply function on plugin.
+        permissions[0] = createPermission(
+            BulkPermissionsLib.Operation.Revoke,
+            PLUGIN_PLACEHOLDER,
+            DAO_PLACEHOLDER,
```

Comment:
It would be good that the example illustrates how to update from a specific past version. But probably better for a later time